src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Serialization/JsonReaderExtensions.cs

```diff
@@ -1,78 +1,35 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
-#nullable disable
-
 using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
 using Newtonsoft.Json;
 
 namespace Microsoft.AspNetCore.Razor.ProjectEngineHost.Serialization;
 
 internal static class JsonReaderExtensions
 {
-    public static bool ReadTokenAndAdvance(this JsonReader reader, JsonToken expectedTokenType, out object value)
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
I eventually need to read something on when this is worth adding or not...

Reply:
It's not worth not adding. :smile: These get called a lot and I want them to inline as much as possible.